Source/Core/Core/PatchEngine.cpp

```diff
@@ -211,6 +224,52 @@ static void ApplyPatches(const std::vector<Patch>& patches)
   }
 }
 
+std::unique_ptr<std::map<u64, std::vector<u8>>> CalculateDiscPatches(const DiscIO::Volume &disc) {
+  if (s_file_patches.empty())
+    return nullptr;
+
+  auto partition = disc.GetGamePartition();
+  const DiscIO::FileSystem *fs = disc.GetFileSystem(partition);
+
+  auto patches = std::make_unique<std::map<u64, std::vector<u8>>>();
+
+  for (const auto& patch : s_file_patches) {
+    if (!patch.active)
+      continue;
+
+    const auto& file = fs->FindFileInfo(patch.path);
+    if (file) {
+      for (const auto& entry : patch.entries) {
+        u64 disc_offset = disc.PartitionOffsetToRawOffset(file->GetOffset(), partition) + entry.address;
```

Comment:
This is wrong for Wii discs, since entry.address doesn't account for the hashes at the beginning of each cluster. (Unless you expect the person writing the patch to correct for the hashes manually?) Patching a Wii disc using the current code is inconvenient anyway, since you would need to handle the encryption manually...

If the code isn't going to support Wii discs properly, you might as well skip the call to `PartitionOffsetToRawOffset`.

Reply:
Disregard the previous comment. You can get Wii discs to work, but you would need to remove this call to `PartitionOffsetToRawOffset`, and `PatchReadRequest` would need to check which partition it's applying patches to.